generator/generator.go

```diff
@@ -408,21 +409,53 @@ func (g *generator) Build(
 		}
 	}
 
-	if !hasPointerStructMethod && ((source.Named && !source.Basic) || (target.Named && !target.Basic) || (source.Pointer && source.PointerInner.Named && !source.PointerInner.Basic)) {
+	isSourceNamed, isTargetNamed := source.Named && !source.Basic, target.Named && !target.Basic
+	isSourcePointerNamed := source.Pointer && source.PointerInner.Named && !source.PointerInner.Basic
+	isTargetPointerNamed := target.Pointer && target.PointerInner.Named && !target.PointerInner.Basic
+	if !hasPointerStructMethod && (isSourceNamed || isTargetNamed || isSourcePointerNamed || isTargetPointerNamed) {
```

Comment:
Why this change? I reverted the 4 lines above and no test failed.

Reply:
Let me think more about it.

Reply:
Got rid of this. Checked by tests.